.github/workflows/test-prs.yml

```diff
@@ -83,45 +89,77 @@ jobs:
           distribution: 'microsoft'
 
       - name: Setup Gradle
-        uses: gradle/gradle-build-action@v3
+        uses: gradle/actions/setup-gradle@v3
 
       - name: Test
-        run: ./gradlew --info -s ${{ matrix.test }}
+        if: ${{ matrix.test.filter != null }}
+        run: ./gradlew --info -s ${{ matrix.test.path }} --tests "${{ matrix.test.filter }}"
+
+      - name: Test
+        if: ${{ matrix.test.filter == null }}
+        run: ./gradlew --info -s ${{ matrix.test.path }}
 
       # Always upload test results
       - name: Merge Test Reports
         if: success() || failure()
         run: npx junit-report-merger junit.xml "**/TEST-*.xml"
 
       - name: Format test run name as artifact name
-        id: format-artifact-name
+        if: (success() || failure()) &&  runner.os == 'Windows'
+        id: format-artifact-name-windows
+        # Use the GITHUB_OUTPUT mechanic to set the output variable
+        run: |
+          # We need to remove all invalid characters from the test name:
+          # Invalid characters include:  Double quote ", Colon :, Less than <, Greater than >, Vertical bar |, Asterisk *, Question mark ?, Carriage return \r, Line feed \n, Backslash \, Forward slash /
+          $NAME = "${{ matrix.test.displayName }}" -replace '[":<>|*?\\/]', '-' -replace ' ', ''
+          
+          # Check if the GITHUB_OUTPUT is set
+          Write-Output "Determined name to be $NAME-${{ matrix.os }}"
+          if ([string]::IsNullOrEmpty($env:GITHUB_OUTPUT)) {
+            # We do not have github output, then use the set output command
+            Write-Output "::set-output name=artifact-name::$NAME-${{ matrix.os }}"
+            exit
+          }
+          Add-Content -Path $env:GITHUB_OUTPUT -Value "artifact-name=$NAME-${{ matrix.os }}"
+
+      - name: Format test run name as artifact name
+        if: (success() || failure()) &&  runner.os != 'Windows'
```

Comment:
While Mac is a *nix system, if you're going to test on it explicitly (which will always take longer considering concurrent runners are limited to 5), it shouldn't be put in the same bucket as Linux.

But I'd prefer if we hadn't tested mac to be honest. It's less used than Windows and Linux (ci), and it's meant to act similarly to Linux for the most part.

Reply:
This has nothing to do with mac vs linux. But with bash vs powershell......

And I think we should test on mac.
It is an OS a significant part of our modders use, and we do not actually use the runners for anything else either.